app/code/core/Mage/Usa/Model/Shipping/Carrier/Ups.php

```diff
@@ -1201,62 +1205,60 @@ protected function _parseRestTrackingResponse($trackingValue, $jsonResponse)
         if ($jsonResponse) {
             $responseData = json_decode($jsonResponse, true);
 
-            if ($responseData['trackResponse']['shipment']) {
+            if (isset($responseData['trackResponse']['shipment'])) {
                 $activityTags = $responseData['trackResponse']['shipment'][0]['package'][0]['activity'] ?? [];
-                if ($activityTags) {
```

Comment:
Taking away this `if` changes the logic, and I'm not sure if it's correct.

If there is no activity, the old code will leave `$resultArr` as an empty array. This new version will set `$resultArr['progressdetail'] = $packageProgress`, no matter what. 

Then when `$resultArr` is passed to `setTrackingResultData`, the `if ($resultArr)` would evaluate to false in the old code, but true in the new code. This means that we will be appending a (empty) tracking-result-status to the tracking result in the new code, where the old code would have an empty tracking result.

So, I think this is incorrect and should be left as it was.